src/test/common/ios/plistBuddy.test.ts

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for details.
+
+import {PlistBuddy} from "../../../common/ios/plistBuddy";
+
+import * as Q from "q";
+
+suite("plistBuddy", function() {
+    suite("commonContext", function() {
+        test("Should attempt to set plist properties correctly", function() {
+            const plistFileName = "testFile.plist";
+            const plistProperty = "myProperty";
+            const plistValue = "myValue";
+
+            const deferred1 = Q.defer<void>();
+            const deferred2 = Q.defer<void>();
+
+            const mockExec: any = {
+                exec: (command: string, opts: any) => {
+                    if (command.match(/Set/)) {

```

Comment:
I'd try to refactor this to make the code more linear/simple. Hopefully it should be obvious to see that a test is bug free, and in this case it's not that simple.